client_it_test.go

```diff
@@ -213,3 +214,95 @@ func TestClient_AddDistributedObjectListener(t *testing.T) {
 		mu.Unlock()
 	})
 }
+
+func TestClusterReconnection(t *testing.T) {
+	ctx := context.Background()
+	cls := it.StartNewClusterWithOptions("go-cli-test-cluster", 15701, 3)
+	mu := &sync.Mutex{}
+	events := []hz.LifecycleState{}
+	config := cls.DefaultConfig()
+	config.AddLifecycleListener(func(event hz.LifecycleStateChanged) {
+		mu.Lock()
+		events = append(events, event.State)
+		mu.Unlock()
+	})
+	c, err := hz.StartNewClientWithConfig(ctx, config)
+	if err != nil {
+		t.Fatal(err)
+	}
+	time.Sleep(5 * time.Second)
+	cls.Shutdown()
+	time.Sleep(5 * time.Second)
+	cls = it.StartNewClusterWithOptions("go-cli-test-cluster", 15701, 3)
+	time.Sleep(5 * time.Second)
+	cls.Shutdown()
+	c.Shutdown(ctx)
+	mu.Lock()
+	defer mu.Unlock()
+	target := []hz.LifecycleState{
+		hz.LifecycleStateStarting,
+		hz.LifecycleStateClientConnected,
```

Comment:
Should we drop the name `Client` from the event names?

I mean `LifecycleStateClientConnected` -> `LifecycleStateConnected`.

We use Client in Java side because member and client use the same enum, but I don't see a reason to do so in our non-Java clients

Reply:
Renamed that at: https://github.com/hazelcast/hazelcast-go-client/pull/590/commits/d19d9f74ec76de7388c57eb1cbe0a5bc1e68a057